typescript/phoenix-sdk/src/market.ts

```diff
@@ -197,104 +372,217 @@ export class Market {
-  let traderBuffer = remaining.subarray(offset, offset + tradersSize);
-
-  let bidsUnsorted = deserializeRedBlackTree(
-    bidBuffer,
-    orderIdBeet,
-    restingOrderBeet
-  );
-  let asksUnsorted = deserializeRedBlackTree(
-    askBuffer,
-    orderIdBeet,
-    restingOrderBeet
-  );
-
-  // TODO: Respect price-time ordering
-  let bids = [...bidsUnsorted].sort(
-    (a, b) => toNum(-a[0].priceInTicks) + toNum(b[0].priceInTicks)
-  );
-
-  // TODO: Respect price-time ordering
-  let asks = [...asksUnsorted].sort(
-    (a, b) => toNum(a[0].priceInTicks) - toNum(b[0].priceInTicks)
-  );
-
-  let traders = new Map<PublicKey, TraderState>();
-  for (const [k, traderState] of deserializeRedBlackTree(
-    traderBuffer,
-    publicKeyBeet,
-    traderStateBeet
-  )) {
-    traders.set(k.publicKey, traderState);
+  /**
+   * Returns a Phoenix swap order packet
+   *
+   * @param market The market to submit the order to
+   * @param side The side of the order
+   * @param inAmount The amount of the input token
+   * @param slippage The slippage tolerance in bps
+   * @param selfTradeBehavior The self trade behavior
+   * @param matchLimit The match limit
+   * @param useOnlyDepositedFunds Whether to use only deposited funds
+   */
+  getSwapOrderPacket({
+    side,
+    inAmount,
+    slippage = DEFAULT_SLIPPAGE_PERCENT,
+    selfTradeBehavior = SelfTradeBehavior.Abort,
+    matchLimit = DEFAULT_MATCH_LIMIT,
+    useOnlyDepositedFunds = false,
+  }: {
+    side: Side;
+    inAmount: number;
+    slippage?: number;
+    selfTradeBehavior?: SelfTradeBehavior;
+    matchLimit?: number;
+    useOnlyDepositedFunds?: boolean;
+  }): Partial<OrderPacket> {
+    const expectedOutAmount = this.getExpectedOutAmount({
+      side,
+      inAmount,
+    });
+    const baseMul = 10 ** this.baseToken.data.decimals;
+    const quoteMul = 10 ** this.quoteToken.data.decimals;
+    const slippageDenom = 1 - slippage;
+    let numBaseLots = 0;
+    let minBaseLotsToFill = 0;
+    let numQuoteLots = 0;
+    let minQuoteLotsToFill = 0;
+
+    if (side === Side.Ask) {
+      numBaseLots =
+        (inAmount * baseMul) /
+        parseFloat(this.data.header.baseLotSize.toString());
+      minQuoteLotsToFill = Math.ceil(
+        ((expectedOutAmount * quoteMul) /
+          parseFloat(this.data.header.quoteLotSize.toString())) *
+          slippageDenom
+      );
+    } else {
+      numQuoteLots =
+        (inAmount * quoteMul) /
+        parseFloat(this.data.header.quoteLotSize.toString());
+      minBaseLotsToFill = Math.ceil(
+        ((expectedOutAmount * baseMul) /
+          parseFloat(this.data.header.baseLotSize.toString())) *
+          slippageDenom
+      );
+    }
+
+    const order: Partial<OrderPacket> = {
+      side,
+      priceInTicks: null, // TODO what is this?
+      numBaseLots,
+      minBaseLotsToFill,
+      numQuoteLots,
+      minQuoteLotsToFill,
+      selfTradeBehavior,
+      matchLimit,
+      clientOrderId: 0, // TODO what is this?
```

Comment:
this is a param that the client can optionally set on their order, if they have an internal order id they want attached to the order on-chain

Reply:
default 0 is correct for sdk

Reply:
yeah i'll add that optionally